src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -485,11 +475,22 @@ const generateTaskRole = (taskRole, labels, config) => {
 };
 
 const generateFrameworkDescription = (frameworkName, virtualCluster, config, rawConfig) => {
+  // calculate pod priority
+  // reference: https://github.com/microsoft/pai/issues/3704
+  let jobPriority = 0;
+  if (launcherConfig.enabledHived) {
+    jobPriority = parseInt(Object.values(config.taskRoles)[0].hivedPodSpec.priority);
+    jobPriority = Math.min(Math.max(jobPriority, -1), 126);
+  }
+  const jobCreationTime = Math.floor(new Date() / 1000) & (Math.pow(2, 23) - 1);
+  const podPriority = - (((126 - jobPriority) << 23) + jobCreationTime);
+
   const [userName, jobName] = frameworkName.split(/~(.+)/);
   const frameworkLabels = {
     jobName,
     userName,
     virtualCluster,
+    priority: `p${podPriority}`,
```

Comment:
Why add priority as label?

Reply:
Just move "calculate pod priority" to put()?

---
In reply to: [337993969](https://github.com/microsoft/pai/pull/3766#discussion_r337993969) [](ancestors = 337993969)